Sources/PointFree/Stripe.swift

```diff
@@ -0,0 +1,220 @@
+    createSubscription: PointFree.createSubscription,
+    fetchCustomer: PointFree.fetchCustomer,
+    fetchPlans: PointFree.fetchPlans,
+    fetchPlan: PointFree.fetchPlan,
+    fetchSubscription: PointFree.fetchSubscription
+  )
+
+  public struct Cents: SingleValueCodable {
+    public let rawValue: Int
+    public init(rawValue: Int) {
+      self.rawValue = rawValue
+    }
+  }
+
+  public struct Customer: Codable {
+    public let id: String
+  }
+
+  public struct Subscription: Codable {
+    let canceledAt: Date?
+    let cancelAtPeriodEnd: Bool
+    let created: Date
+    let currentPeriodStart: Date? // TODO: Audit nullability
+    let currentPeriodEnd: Date? // TODO: Audit nullability
+    let customer: String
+    let endedAt: Date?
+    let id: String
+    let plan: Plan
+    let quantity: Int
+    let start: Date
+    let status: Status
+
+    private enum CodingKeys: String, CodingKey {
+      case canceledAt = "canceled_at"
+      case cancelAtPeriodEnd = "cancel_at_period_end"
+      case customer
+      case created
+      case currentPeriodEnd = "current_period_end"
+      case currentPeriodStart = "current_period_start"
+      case endedAt = "ended_at"
+      case id
+      case plan
+      case quantity
+      case start
+      case status
+    }
+
+    public enum Status: String, Codable {
+      case trialing
+      case active
+      case pastDue = "past_due"
+      case canceled
+      case unpaid
+    }
+  }
+
+  public struct Plan: Codable {
+    let amount: Cents
+    let created: Date
+    let currency: Currency
+    let id: Id
+    let interval: Interval
+    let metadata: [String: String]
+    let name: String
+    let statementDescriptor: String?
+
+    private enum CodingKeys: String, CodingKey {
+      case amount
+      case created
+      case currency
+      case id
+      case interval
+      case metadata
+      case name
+      case statementDescriptor = "statement_descriptor"
+    }
+
+    public enum Interval: String, Codable {
+      case month
+      case year
+    }
+
+    public enum Currency: String, Codable {
+      case usd
+    }
+
+    public enum Id: String, Codable, RawRepresentable {
+      case yearly
+      case monthly
+      case yearlyTeam = "yearly-team"
+      case monthlyTeam = "monthly-team"
+
+      static let all: [Id] = [.yearly, .monthly, .yearlyTeam, .monthlyTeam]
+    }
+  }
+
+  public struct PlansEnvelope: Codable {
+    let data: [Plan]
+    let hasMore: Bool
+
+    private enum CodingKeys: String, CodingKey {
+      case data
+      case hasMore = "has_more"
+    }
+  }
+
+  public struct Token: Codable {
+    let id: String
+  }
+}
+
+//private let subscriptionPlans = fetchPlans
+//  .map(^\.data >>> filter(StripeSubscriptionPlan.Id.all.contains <<< ^\.id))
+
+private func cancelSubscription(id: String) -> EitherIO<Prelude.Unit, Stripe.Subscription> {
+  return stripeDataTask("https://api.stripe.com/v1/subscriptions/\(id)", .delete)
```

Comment:
I forget, is stripe sandbox api entirely determined by the api keys, or is there another URL to hit?